### PR TITLE
UINV-33 Accounting code filter list populated based on vendor selected

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Accounting code filter list populated based on vendor selected
https://issues.folio.org/browse/UINV-33
## Approach
Collect all selected vendor accounting codes (fields: `erpCode` and `appSystemNo`) to create list of options
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 1]
```diff
@@ -115,6 +121,10 @@ class InvoiceForm extends Component {
     const isEditPostApproval = IS_EDIT_POST_APPROVAL(initialValues.id, initialValues.status);
     const metadata = initialValues.metadata;
     const approvedBy = get(initialValues, 'approvedBy');
+    const selectedVendor = find(orgs, { id: get(formValues, 'vendorId') });
+    const vendorCode = get(selectedVendor, 'erpCode');
+    const vendorAccountCodes = selectedVendor ? get(selectedVendor, 'accounts', []).map(account => get(account, 'appSystemNo')) : [];
+    const vendorCodes = uniq([vendorCode, ...vendorAccountCodes].filter(Boolean));
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceForm/InvoiceForm.js]: IMHO, these 3 lines look like a function, `getAccountCodes(organization)`.

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 2]
```diff
@@ -302,12 +312,11 @@ class InvoiceForm extends Component {
...
                       <Col data-test-col-accounting-code xs={3}>
                         <FieldSelection
-                          dataOptions={[]}
-                          disabled
+                          dataOptions={getCodeOptions(vendorCodes)}
                           label={<FormattedMessage id="ui-invoice.invoice.accountingCode" />}
                           name="accountingCode"
                           required
-                        // validate={validateRequired}
+                          // validate={validateRequired}
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceForm/InvoiceForm.js]: this change doesn't make too much sense. Removing would make more sense, or no changes to this line

Comment (AndreiShumski2) [reply src/invoices/InvoiceForm/InvoiceForm.js]: Let's left this line as is. For now `accounting code` is not required field in the organization form

File: src/invoices/InvoiceForm/InvoiceForm.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/invoices/InvoiceForm/InvoiceForm.js]: Updated

File: src/common/utils/getAccountCodes.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { get, uniq } from 'lodash';
+
+// eslint-disable-next-line import/prefer-default-export
+export const getAccountCodes = (vendor) => {
```
Comment (NikitaSedyx) [inline src/common/utils/getAccountCodes.js]: is it common util?

Comment (AndreiShumski2) [reply src/common/utils/getAccountCodes.js]: I think so. I'm going to use it for `InvoiceLineForm` too.

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 4]
```diff
@@ -115,6 +121,8 @@ class InvoiceForm extends Component {
     const isEditPostApproval = IS_EDIT_POST_APPROVAL(initialValues.id, initialValues.status);
     const metadata = initialValues.metadata;
     const approvedBy = get(initialValues, 'approvedBy');
+    const selectedVendor = find(orgs, { id: get(formValues, 'vendorId') });
+    const accountCodes = selectedVendor ? getAccountCodes(selectedVendor) : [];
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceForm/InvoiceForm.js]: wouldn't it be more proper and handy to manage this case inside of `getAccountCodes`? So you don't have to copy-paste that code in every place you want to get `accountCodes`

Comment (AndreiShumski2) [reply src/invoices/InvoiceForm/InvoiceForm.js]: Right suggestion. Thank you

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 5]
```diff
@@ -302,8 +310,7 @@ class InvoiceForm extends Component {
                       </Col>
                       <Col data-test-col-accounting-code xs={3}>
                         <FieldSelection
-                          dataOptions={[]}
-                          disabled
+                          dataOptions={getCodeOptions(accountCodes)}
                           label={<FormattedMessage id="ui-invoice.invoice.accountingCode" />}
                           name="accountingCode"
                           required
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceForm/InvoiceForm.js]: it's not required field, AFAIK

Comment (aliaksei-chumakou) [reply src/invoices/InvoiceForm/InvoiceForm.js]: also `validate` property could be removed

File: src/invoices/InvoiceForm/InvoiceForm.js [hunk 6]
```diff
@@ -302,8 +310,7 @@ class InvoiceForm extends Component {
                       </Col>
                       <Col data-test-col-accounting-code xs={3}>
                         <FieldSelection
-                          dataOptions={[]}
-                          disabled
+                          dataOptions={getCodeOptions(accountCodes)}
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceForm/InvoiceForm.js]: Basing on the last comment from Dennis, labels of options should look like `acc number (acc code)`, please check that https://issues.folio.org/browse/UINV-36?focusedCommentId=56485&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-56485

Comment (AndreiShumski2) [reply src/invoices/InvoiceForm/InvoiceForm.js]: In this case, we'll have some problems when a vendor has acc number (as a required field) but hasn't got any acc code -> the option will look like ` acNumber123()`. So if a user selects this option we'll send to back-end empty string

File: src/invoices/InvoiceForm/InvoiceForm.js (hunk 5 above)
Comment (AndreiShumski2) [reply src/invoices/InvoiceForm/InvoiceForm.js]: fixed

File: src/invoices/InvoiceForm/InvoiceForm.js (hunk 6 above)
Comment (aliaksei-chumakou) [reply src/invoices/InvoiceForm/InvoiceForm.js]: this field isn't required, I believe this should be fine

**Pull request merged**